corehq/apps/data_interfaces/tasks.py

```diff
@@ -192,3 +192,10 @@ def run_case_update_rules_on_save(case):
                 AutomaticUpdateRule.WORKFLOW_CASE_UPDATE).filter(case_type=case.type)
             now = datetime.utcnow()
             run_rules_for_case(case, rules, now)
+
+
+@periodic_task(run_every=crontab(hour=0, minute=0), queue='case_rule_queue', ignore_result=True)
+def delete_old_rule_submission_logs():
+    start = datetime.utcnow()
+    max_age = start - timedelta(days=90)
+    CaseRuleSubmission.objects.filter(created_on__lt=max_age).delete()
```

Comment:
I kept this very simple since deleting these is fast as there's not much data and it's just a plain SQL delete. I didn't think there was a need for any kind of batching or anything like that.